File: cpp/recursion/tower_of_hanoi.cpp
//Tower of Hanoi Problem using recursion

#include<iostream>
using namespace std;

void TOH(int n, int A, int B, int C){
    if (n>0)
    {
        TOH(n-1, A, C, B);
        cout << "disk moved from " << A << " to " << B<<endl;
        cout << "(" << A <<", "<< B << ")" <<endl;
        TOH(n-1, B,A,C);
    }
}

int main()
{
    TOH(3,1,2,3);
    return 0;
}

File: python/recursion/tower_of_hanoi.py
# Tower of Hanoi Problem using recursion

def TOH(n, A, B, C):
    if n>0:
        TOH(n-1, A, C, B)
        print("disk moved from " + str(A) +" to "+str(B))
        print("(" +str(A) +", "+ str(B) +")")
        TOH(n-1, B,A,C)

TOH(3,1,2,3)
